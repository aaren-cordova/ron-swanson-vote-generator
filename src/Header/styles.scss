@import 'styles/module';

.Header {
  display: flex;
  flex-direction: row;
  justify-content: space-between
}

.Header__section {
  display: flex;
  flex-direction: row;
}

.Header__section:last-child {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 500px;
}

.Header__title {
  font-weight: 700;
  margin-right: rem(8px);
}

.Header__subtitle {
}