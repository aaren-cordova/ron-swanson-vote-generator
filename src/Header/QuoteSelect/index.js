import {Select} from 'antd';
import './styles.scss';
import classNames from 'classnames';
import compareQuotes from './compareQuotes';

const {Option} = Select;

export default function QuoteSelect({className, quotes, setSearchTerm}) {
    function onSearch(searchTerm) {
        setSearchTerm(searchTerm);
    }

    function onChange(event) {
        console.log('change', event)
    }


    return (
        <Select
            className={classNames('QuoteSelect', className)}
            size='small'
            defalutValue
            labelInValue
            allowClear
            autoClearSearchValue={false}
            showSearch
            placeholder="Select a quote"
            optionFilterProp="label"
            onSearch={onSearch}
            onChange={onChange}
            options={quotes}
            disabled={!quotes.length}
            mode='tag'
            filterOption={compareQuotes}
        >
            {quotes.map(({value, label}) => <Option value={value} key={value}>{label}</Option>)}
        </Select>
    );
}
