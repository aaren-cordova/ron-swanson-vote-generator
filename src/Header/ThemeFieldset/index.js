import classNames from 'classnames';
import './styles.scss';

export default function ThemeFieldset({className, theme, setTheme}) {
    function onChange(event) {
        setTheme(event.target.value);
    }

    return (
        <fieldset
            className={classNames('ThemeFieldset', className)}
            onChange={onChange}
        >
            <label className="ThemeFieldset__label">Theme</label>
            <input
                type="radio"
                id="theme-light-input"
                name="theme"
                value="light"
                checked={theme === 'light'}
            />
            <label
                className="ThemeFieldset__label"
                htmlFor="theme-light-input"
            >
                Light
            </label>
            <input
                type="radio"
                id="theme-dark-input"
                name="theme"
                value="dark"
                checked={theme === 'dark'}
            />
            <label
                className="ThemeFieldset__label"
                htmlFor="theme-dark-input"
            >
                Dark
            </label>
        </fieldset>
    );
}