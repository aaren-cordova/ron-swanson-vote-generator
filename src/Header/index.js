import './styles.scss';
import classNames from 'classnames';


export default function Header(
    {
        className,
        quotes,

    },
) {

    return (
        <header className={classNames('Header', className)}>
            <section className="Header__section">
                <h3 className="Header__title">
                    Ron Swanson Voter
                    <span className="Header__votes">({quotes.length})</span>
                </h3>
                <h4 className="Header__subtitle">&quot;Vote for your favorite quote!&quot;</h4>
            </section>


        </header>
    );
}
