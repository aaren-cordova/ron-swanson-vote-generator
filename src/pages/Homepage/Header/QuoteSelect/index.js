import {Select} from 'antd';
import './styles.scss';
import classNames from 'classnames';
import compareQuotes from './compareQuotes';
import PropTypes from 'prop-types';
import {UNDEFINED_VOTE, YES_VOTE} from '../../../../constants';
const {Option} = Select;

QuoteSelect.propTypes = {
    className: PropTypes.string,
    quotes: PropTypes.array.isRequired,
    setSearchTerm: PropTypes.func.isRequired,
    setCurrentVote: PropTypes.func.isRequired,

};

export default function QuoteSelect({className, quotes, setSearchTerm, setCurrentVote}) {
    function handleSearch(searchTerm) {
        setSearchTerm(searchTerm);
    }

    function handleChange({value: id}) {
        setCurrentVote({id, value: YES_VOTE});
    }

    return (
        <Select
            className={classNames('QuoteSelect', className)}
            allowClear
            autoClearSearchValue={false}
            disabled={!quotes.length}
            filterOption={compareQuotes}
            labelInValue
            mode='tag'
            onSearch={handleSearch}
            onChange={handleChange}
            optionFilterProp="label"
            options={quotes.map(({value, label}) => ({value, label} ))}
            placeholder="Select a quote"
            showSearch
            size='small'
        >
            {quotes.map(({value, label}) => <Option value={value} key={value}>{label}</Option>)}
        </Select>
    );
}
