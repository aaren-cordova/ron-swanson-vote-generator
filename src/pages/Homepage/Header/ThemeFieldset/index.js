import classNames from 'classnames';
import './styles.scss';
import PropTypes from 'prop-types';
import {Radio} from 'antd';

ThemeFieldset.propTypes = {
    className: PropTypes.string,
    setTheme: PropTypes.func.isRequired,
    theme: PropTypes.string.isRequired,
};
export default function ThemeFieldset({className, theme, setTheme}) {
    function onChange(event) {
        setTheme(event.target.value);
    }

    return (
        <fieldset
            className={classNames('ThemeFieldset', className)}
        >
            <label className="ThemeFieldset__label">Theme</label>

            <Radio.Group
                onChange={onChange}
                value={theme}
            >
                <Radio
                    id="theme-light-input"
                    name="theme"
                    value="light"
                />
                <label
                    className="ThemeFieldset__label"
                    htmlFor="theme-light-input"
                >
                    Light
                </label>
                <Radio
                    id="theme-dark-input"
                    name="theme"
                    value="dark"
                />
                <label
                    className="ThemeFieldset__label"
                    htmlFor="theme-dark-input"
                >
                    Dark
                </label>
            </Radio.Group>


        </fieldset>
    );
}