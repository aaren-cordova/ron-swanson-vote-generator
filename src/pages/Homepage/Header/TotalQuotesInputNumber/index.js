import {InputNumber} from 'antd';
import classNames from 'classnames';
import './styles.scss';
import PropTypes from 'prop-types';

TotalQuotesInputNumber.propTypes = {
    className: PropTypes.string,
    totalQuotes: PropTypes.number.isRequired,
    setTotalQuotes: PropTypes.func.isRequired,
};

export default function TotalQuotesInputNumber({className, totalQuotes, setTotalQuotes}) {
    function handleTotalQuotesChange(totalQuotes) {
        setTotalQuotes(totalQuotes);
    }

    return (
        <div className={classNames('TotalQuotesInputNumber', className)}>
            <label
                className="TotalQuotesInputNumber__label"
                htmlFor="total-quotes-input-number"
            >
                Total Quotes
            </label>
            <InputNumber
                id="total-quotes-input-number"
                onChange={handleTotalQuotesChange}
                size={'small'}
                min={2}
                max={100}
                step={1}
                value={totalQuotes}
            />
        </div>
    );
}