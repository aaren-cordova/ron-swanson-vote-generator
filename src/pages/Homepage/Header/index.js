import './styles.scss';
import classNames from 'classnames';
import ThemeFieldset from './ThemeFieldset';
import QuoteSelect from './QuoteSelect';
import TotalQuotesInputNumber from './TotalQuotesInputNumber';
import {useMemo} from 'react';
import PropTypes from 'prop-types';

Header.propTypes = {
    className: PropTypes.string,
    quotes: PropTypes.array.isRequired,
    filteredQuotes: PropTypes.array.isRequired,
    setSearchTerm: PropTypes.func.isRequired,
    setTheme: PropTypes.func.isRequired,
    setTotalQuotes: PropTypes.func.isRequired,
    theme: PropTypes.string.isRequired,
    totalQuotes: PropTypes.number.isRequired,
    setCurrentVote: PropTypes.func.isRequired,
};

export default function Header(
    {
        className,
        quotes,
        filteredQuotes,
        theme,
        setTheme,
        totalQuotes,
        setCurrentVote,
        setTotalQuotes,
        setSearchTerm,
    },
) {
    const totalVotes = useMemo(() => filteredQuotes.reduce((accumulator, {numYesVotes, numNoVotes}) => {
        return accumulator + numYesVotes + numNoVotes;
    }, 0), [filteredQuotes]);

    return (
        <header className={classNames('Header', className)}>
            <section className="Header__section">
                <h3 className="Header__title">
                    Ron Swanson Voter
                </h3>
                <h4 className="Header__subtitle">&quot;Vote for your favorite quote!&quot;</h4>
            </section>
            <section className="Header__section">
                <TotalQuotesInputNumber
                    totalQuotes={totalQuotes}
                    setTotalQuotes={setTotalQuotes}
                />
                <ThemeFieldset
                    theme={theme}
                    setTheme={setTheme}
                />
                <QuoteSelect
                    quotes={quotes}
                    setSearchTerm={setSearchTerm}
                    setCurrentVote={setCurrentVote}
                />

                <span className="Header__votes">Total Votes: {totalVotes}</span>
            </section>

        </header>
    );
}
