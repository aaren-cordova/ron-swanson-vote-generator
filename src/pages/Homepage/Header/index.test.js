import renderer from 'react-test-renderer';
import Header from './index';
import PropTypes from 'prop-types';


let props;

function render() {
    return <Header {...props}/>;
}

beforeEach(() => {
    props = {
        quotes: [],
        filteredQuotes: [],
        theme: '',
        totalQuotes: 0,
        setSearchTerm: jest.fn(),
        setTheme: jest.fn(),
        setTotalQuotes: jest.fn(),
        setCurrentVote: jest.fn(),
    };

});

test('matches the snapshot', () => {
    let component = renderer.create(render());
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('@className="some-class-name" matches the snapshot', () => {
    props.className = 'some-class-name';
    let component = renderer.create(render());
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
