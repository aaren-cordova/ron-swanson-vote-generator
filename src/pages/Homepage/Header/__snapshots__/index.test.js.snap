// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@className="some-class-name" matches the snapshot 1`] = `
<header
  className="Header some-class-name"
>
  <section
    className="Header__section"
  >
    <h3
      className="Header__title"
    >
      Ron Swanson Voter
    </h3>
    <h4
      className="Header__subtitle"
    >
      "Vote for your favorite quote!"
    </h4>
  </section>
  <section
    className="Header__section"
  >
    <div
      className="TotalQuotesInputNumber"
    >
      <label
        className="TotalQuotesInputNumber__label"
        htmlFor="total-quotes-input-number"
      >
        Total Quotes
      </label>
      <div
        className="ant-input-number ant-input-number-sm ant-input-number-out-of-range"
        onBlur={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <div
          className="ant-input-number-handler-wrap"
        >
          <span
            aria-disabled={false}
            aria-label="Increase Value"
            className="ant-input-number-handler ant-input-number-handler-up"
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            role="button"
            unselectable="on"
          >
            <span
              aria-label="up"
              className="anticon anticon-up ant-input-number-handler-up-inner"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="up"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                />
              </svg>
            </span>
          </span>
          <span
            aria-disabled={true}
            aria-label="Decrease Value"
            className="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            role="button"
            unselectable="on"
          >
            <span
              aria-label="down"
              className="anticon anticon-down ant-input-number-handler-down-inner"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          className="ant-input-number-input-wrap"
        >
          <input
            aria-valuemax={100}
            aria-valuemin={2}
            aria-valuenow="0"
            autoComplete="off"
            className="ant-input-number-input"
            id="total-quotes-input-number"
            onChange={[Function]}
            role="spinbutton"
            step={1}
            value="0"
          />
        </div>
      </div>
    </div>
    <fieldset
      className="ThemeFieldset"
    >
      <label
        className="ThemeFieldset__label"
      >
        Theme
      </label>
      <div
        className="ant-radio-group ant-radio-group-outline"
      >
        <label
          className="ant-radio-wrapper"
        >
          <span
            className="ant-radio"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-radio-input"
              id="theme-light-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              type="radio"
              value="light"
            />
            <span
              className="ant-radio-inner"
            />
          </span>
        </label>
        <label
          className="ThemeFieldset__label"
          htmlFor="theme-light-input"
        >
          Light
        </label>
        <label
          className="ant-radio-wrapper"
        >
          <span
            className="ant-radio"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-radio-input"
              id="theme-dark-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              type="radio"
              value="dark"
            />
            <span
              className="ant-radio-inner"
            />
          </span>
        </label>
        <label
          className="ThemeFieldset__label"
          htmlFor="theme-dark-input"
        >
          Dark
        </label>
      </div>
    </fieldset>
    <div
      className="ant-select ant-select-sm QuoteSelect ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-disabled ant-select-show-search"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
    >
      <div
        className="ant-select-selector"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <span
          className="ant-select-selection-search"
        >
          <input
            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
            aria-autocomplete="list"
            aria-controls="rc_select_TEST_OR_SSR_list"
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="rc_select_TEST_OR_SSR_list"
            autoComplete="off"
            className="ant-select-selection-search-input"
            disabled={true}
            id="rc_select_TEST_OR_SSR"
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            readOnly={false}
            role="combobox"
            style={
              Object {
                "opacity": null,
              }
            }
            type="search"
            unselectable={null}
            value=""
          />
        </span>
        <span
          className="ant-select-selection-placeholder"
        >
          Select a quote
        </span>
      </div>
      <span
        aria-hidden={true}
        className="ant-select-arrow"
        onMouseDown={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <span
          aria-label="down"
          className="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
    <span
      className="Header__votes"
    >
      Total Votes: 
      0
    </span>
  </section>
</header>
`;

exports[`matches the snapshot 1`] = `
<header
  className="Header"
>
  <section
    className="Header__section"
  >
    <h3
      className="Header__title"
    >
      Ron Swanson Voter
    </h3>
    <h4
      className="Header__subtitle"
    >
      "Vote for your favorite quote!"
    </h4>
  </section>
  <section
    className="Header__section"
  >
    <div
      className="TotalQuotesInputNumber"
    >
      <label
        className="TotalQuotesInputNumber__label"
        htmlFor="total-quotes-input-number"
      >
        Total Quotes
      </label>
      <div
        className="ant-input-number ant-input-number-sm ant-input-number-out-of-range"
        onBlur={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <div
          className="ant-input-number-handler-wrap"
        >
          <span
            aria-disabled={false}
            aria-label="Increase Value"
            className="ant-input-number-handler ant-input-number-handler-up"
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            role="button"
            unselectable="on"
          >
            <span
              aria-label="up"
              className="anticon anticon-up ant-input-number-handler-up-inner"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="up"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                />
              </svg>
            </span>
          </span>
          <span
            aria-disabled={true}
            aria-label="Decrease Value"
            className="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            role="button"
            unselectable="on"
          >
            <span
              aria-label="down"
              className="anticon anticon-down ant-input-number-handler-down-inner"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          className="ant-input-number-input-wrap"
        >
          <input
            aria-valuemax={100}
            aria-valuemin={2}
            aria-valuenow="0"
            autoComplete="off"
            className="ant-input-number-input"
            id="total-quotes-input-number"
            onChange={[Function]}
            role="spinbutton"
            step={1}
            value="0"
          />
        </div>
      </div>
    </div>
    <fieldset
      className="ThemeFieldset"
    >
      <label
        className="ThemeFieldset__label"
      >
        Theme
      </label>
      <div
        className="ant-radio-group ant-radio-group-outline"
      >
        <label
          className="ant-radio-wrapper"
        >
          <span
            className="ant-radio"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-radio-input"
              id="theme-light-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              type="radio"
              value="light"
            />
            <span
              className="ant-radio-inner"
            />
          </span>
        </label>
        <label
          className="ThemeFieldset__label"
          htmlFor="theme-light-input"
        >
          Light
        </label>
        <label
          className="ant-radio-wrapper"
        >
          <span
            className="ant-radio"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-radio-input"
              id="theme-dark-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              type="radio"
              value="dark"
            />
            <span
              className="ant-radio-inner"
            />
          </span>
        </label>
        <label
          className="ThemeFieldset__label"
          htmlFor="theme-dark-input"
        >
          Dark
        </label>
      </div>
    </fieldset>
    <div
      className="ant-select ant-select-sm QuoteSelect ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-disabled ant-select-show-search"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
    >
      <div
        className="ant-select-selector"
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <span
          className="ant-select-selection-search"
        >
          <input
            aria-activedescendant="undefined_list_0"
            aria-autocomplete="list"
            aria-controls="undefined_list"
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="undefined_list"
            autoComplete="off"
            className="ant-select-selection-search-input"
            disabled={true}
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            readOnly={false}
            role="combobox"
            style={
              Object {
                "opacity": null,
              }
            }
            type="search"
            unselectable={null}
            value=""
          />
        </span>
        <span
          className="ant-select-selection-placeholder"
        >
          Select a quote
        </span>
      </div>
      <span
        aria-hidden={true}
        className="ant-select-arrow"
        onMouseDown={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <span
          aria-label="down"
          className="anticon anticon-down ant-select-suffix"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </span>
    </div>
    <span
      className="Header__votes"
    >
      Total Votes: 
      0
    </span>
  </section>
</header>
`;
